### ISSUE TITLE
[a11y]: AILabel has a11y Violation - Accessible name does not match or contain the visible label text

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#18204]
### Package

@carbon/react

### Package version

1.72.0

### Automated testing tool and ruleset

 IBM Accessibility Checker version 3.1.75 onwards till latest deployment 

### Description

Checker throws _Accessible name does not match or contain the visible label text_  Violation on [AILabel component ]

Violation only seen on  IBM Accessibility Checker version 3.1.75 and later versions (seen on latest deployment as well )
([Here](https://deploy-preview-17351--v11-carbon-react.netlify.app/iframe.html?args=&globals=&id=components-ailabel--default&viewMode=story))

Visually, the button displays two `spans`: "`AI`" and "`Text goes here`"
The aria-label on button is set to "`AI Text goes here`" (with a space)

The space between "AI" and "Text goes here" creates a discrepancy between the visual text and the accessible name.
aria-label="AI Text goes here" -> removing the space in label fixes the violation 

https://deploy-preview-17351--v11-carbon-react.netlify.app/iframe.html?args=&globals=&id=components-ailabel--default&viewMode=story
![Image](https://github.com/user-attachments/assets/0b481a97-5d8b-4507-9e7f-58b5e96898d3)

### Reproduction/example

https://deploy-preview-17351--v11-carbon-react.netlify.app/iframe.html?args=&globals=&id=components-ailabel--default&viewMode=story

https://react.carbondesignsystem.com/iframe.html?args=&globals=&id=components-ailabel--default&viewMode=story

### Steps to reproduce

visit expandable AILabel storybook - https://react.carbondesignsystem.com/iframe.html?args=&globals=&id=components-ailabel--default&viewMode=story
run Accessibility Checker with 3.1.75 and later versions 

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by philljenkins** [comment]
I'm not able to reproduce using the following:
- Page tested: https://react.carbondesignsystem.com/?path=/story/components-tabs--playground 
- Browser: Chrome
- Checker: [v 3.1.78](https://github.com/IBMa/equal-access/releases) (not 3.1.75 as reported above)
- Settings: `Preview Rules` and `IBM Accessibility 7.3`

**Comment by philljenkins** [comment]
my concern was why the `aria-label=` was being redundantly used? 
The text is already visible and set as the "accessible name" on the button for everyone without the attribute.
